web_widget_slickroom/static/tests/js/web_widget_slickroom.js

```diff
@@ -0,0 +1,179 @@
+/* Copyright 2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). */
+
+odoo.define_section('web_widget_slickroom', ['web.core', 'web.form_common'], function (test) {
+    "use strict";
+
+    function appendWidget (core, formCommon, $fix) {
+        var fieldManager = new formCommon.DefaultFieldManager(null, {});
+        var node = {'attrs': {}};
+        var FieldSlickroomImages = core.form_widget_registry.get('slickroom');
+        var widget = new FieldSlickroomImages(fieldManager, node);
+        widget.appendTo($fix);
+        return widget;
+    }
+
+    function imgHTML (id, attr) {
+        return $(
+            '<div><img data-record-id="' +id + '" ' + attr +
+            '="/web/image/ir.attachment/' + id + '/datas"></div>'
+        );
+    }
+
+    test('._openModal() should open a darkroom modal with provided options',
+        function (assert, core, formCommon) {
+            var $fix = $('#qunit-fixture');
+            var widget = appendWidget(core, formCommon, $fix);
+            var recordId = 1;
+            widget.$slick.slick('slickAdd', imgHTML(recordId, 'src'));
+
+            var modalAction = {};
+            widget.do_action = function (action, options) {
+                modalAction = action;
+            };
+
+            var expectedAction = {
+                "type": "ir.actions.act_window",
+                "res_model": "darkroom.modal",
+                "name": "Darkroom",
+                "views": [[false, "form"]],
+                "target": "new",
+                "context": {
+                    "active_field": widget.options.fieldName,
+                    "active_model": widget.options.modelName,
+                    "active_record_id": recordId
+                }
+            };
+
+            widget.$('img').click();
+
+            assert.deepEqual(modalAction, expectedAction);
+        }
+    );
+
+    test('._openModal() should open a darkroom modal with on_close action ' +
+         'that calls ._updateImage()',
+        function (assert, core, formCommon) {
+            var $fix = $('#qunit-fixture');
+            var widget = appendWidget(core, formCommon, $fix);
+            var recordId = 1;
+            widget.$slick.slick('slickAdd', imgHTML(recordId, 'src'));
+
+            var modalOptions = {};
+            widget.do_action = function (action, options) {
+                modalOptions = options;
+            };
+
+            var $img = widget.$('img');
+            $img.click();
+            modalOptions.on_close();
+
+            assert.notStrictEqual($img.attr('src').indexOf('?unique'), -1);
```

Comment:
Instead of testing for an outcome of the call to ``_updateImage``, which you're doing in the ``_updateImage`` test cases anyway, I would mock out the method and just confirm that it was called. You should be able to do this the same way that you modified ``do_action``.